VMEncryption/main/oscrypto/mariner_10/encryptstates/PatchBootSystemState.py

```diff
@@ -0,0 +1,200 @@
+                                      '/etc/dracut.conf.d/ade.conf')
+
+        # Add the new kernel param
+        additional_params = ["rd.luks.ade.partuuid={0}".format(root_partuuid),
+                             "rd.luks.ade.bootuuid={0}".format(boot_uuid),
+                             "rd.debug"]
+        self._add_kernelopts(additional_params)
+
+        # For clarity after reboot, we should also add the correct info to crypttab
+        crypt_item = CryptItem()
+        crypt_item.dev_path = os.path.join("/dev/disk/by-partuuid/", root_partuuid)
+        crypt_item.mapper_name = CommonVariables.osmapper_name
+        crypt_item.luks_header_path = "/boot/luks/osluksheader"
+        self.crypt_mount_config_util.add_crypt_item(crypt_item)
+
+    def _add_kernelopts(self, args_to_add):
+        """
+        For EFI machines (Gen2) we want to use the EFI grub.cfg path
+        For BIOS machines (Gen1) we want to use the old grub.cfg path
+        But we can't tell at this stage easily which one to use if both are present. so we will just update both.
+        Moreover, in case somebody runs grub2-mkconfig on the machine we don't want the changes to get nuked out, we will update grub defaults file too.
+        """
+        grub_cfg_paths = [
+            "/boot/grub2/grub.cfg",
+            "/boot/efi/EFI/redhat/grub.cfg"
+        ]
+        grub_cfg_paths = filter(os.path.exists, grub_cfg_paths)
+
+        for grub_cfg_path in grub_cfg_paths:
+            for arg in args_to_add:
+                self.command_executor.ExecuteInBash("grubby --args {0} --update-kernel DEFAULT -c {1}".format(arg, grub_cfg_path))
+
+        self._append_contents_to_file('\nGRUB_CMDLINE_LINUX+="{0}"\n'.format(" ".join(args_to_add)),
+                                      '/etc/default/grub')
+
+    def _get_kernelopts(self):
+        proc_comm = ProcessCommunicator()
+        self.command_executor.ExecuteInBash("grub2-editenv - list | grep kernelopts", communicator=proc_comm)
+        return proc_comm.stdout.strip()
+
+    def _get_root_partuuid(self):
+        root_partuuid = None
+        root_device_items = self.disk_util.get_device_items(self.rootfs_block_device)
+        for root_item in root_device_items:
+            if self.rootfs_sdx_path.endswith(root_item.name) or os.path.realpath(self.rootfs_block_device).endswith(root_item.name):
+                root_partuuid = self.disk_util.get_device_items_property(root_item.name, "PARTUUID")
+                if root_partuuid:
+                    return root_partuuid
+        return root_partuuid
+
+    def _get_boot_uuid(self):
+        return self._parse_uuid_from_fstab('/boot')
+
+    def _get_luks_uuid(self):
+        luks_header_path = "/boot/luks/osluksheader"
+        return self.disk_util.luks_get_uuid(luks_header_path)
+
+    def _is_detached_header_fix(self):
+        # TODO: We need to ask the redhat folks the best way to do this. We can scan man pages like below.
```

Comment:
please include me on the email thread

Reply:
I copied this code from the RHEL8 `PatchBootSystemState.py` file. What implementation did ADE have in mind?